lib/perl5db.pl

```diff
@@ -4455,13 +4460,13 @@ sub DB::sub {
         # Whether or not the autoloader was running, a scalar to put the
         # sub's return value in (if needed), and an array to put the sub's
         # return value in (if needed).
-        if ($sub eq 'threads::new' && $ENV{PERL5DB_THREADED}) {
+        if (ref $sub && $sub eq 'threads::new' && $ENV{PERL5DB_THREADED}) {
```

Comment:
Isn't this backwards? We only want to check for `'threads::new'` if `$sub` is not a ref.

Reply:
Yep. Fixed.